src/hotspot/share/utilities/resourceHash.hpp

```diff
@@ -210,6 +215,32 @@ class ResourceHashtableBase : public STORAGE {
       ++bucket;
     }
   }
+
+  // ITER contains bool do_entry(K const&, V const&), which will be
+  // called for each entry in the table.  If do_entry() returns true,
+  // the entry is deleted.
+  template<class ITER>
+  void unlink(ITER* iter) {
+    const unsigned sz = table_size();
+    for (unsigned index = 0; index < sz; index++) {
+      Node** ptr = bucket_at(index);
+      while (*ptr != NULL) {
+        Node* node = *ptr;
+        // do_entry must clean up the key and value in Node.
+        bool clean = iter->do_entry(node->_key, node->_value);
```

Comment:
Unless I miss it, you don't enforce the constness here, no? So, nothing prevents the called hook from modifying key or value. You may do what Ioi did in https://github.com/openjdk/jdk/pull/4942 and const-cast the parameters for the call.

Reply:
I thought I answered this in email but I don't see it.  I want the do_entry function to modify the arguments, like calling their destructor or free operators.